utils/gyb_syntax_support/AttributeNodes.py

```diff
@@ -267,19 +263,49 @@
              Child('Label', kind='IdentifierToken',
                    text_choices=['jvp', 'vjp']),
              Child('Colon', kind='ColonToken'),
-             Child('DeclBaseName', kind='Syntax', description='''
+             Child('FunctionDeclName', kind='FunctionDeclName',
+                   description='The referenced function name.'),
+             Child('TrailingComma', kind='CommaToken', is_optional=True),
+         ]),
+
+    # func-decl-name -> (identifier | operator) decl-name-arguments?
+    # NOTE: This is duplicated with `DeclName` above. Change `DeclName`
+    # description and use it if possible.
+    Node('FunctionDeclName', kind='Syntax',
+         description='A function declaration name (e.g. `foo(_:_:)`).',
+         children=[
+             Child('Name', kind='Syntax', description='''
                    The base name of the referenced function.
                    ''',
                    node_choices=[
                        Child('Identifier', kind='IdentifierToken'),
-                       Child('Operator', kind='PrefixOperatorToken'),
+                       Child('PrefixOperator', kind='PrefixOperatorToken'),
+                       Child('SpacedBinaryOperator',
+                             kind='SpacedBinaryOperatorToken'),
                    ]),
-             Child('DeclNameArguments', kind='DeclNameArguments',
+             Child('Arguments', kind='DeclNameArguments',
                    is_optional=True, description='''
                    The argument labels of the referenced function, optionally \
                    specified.
                    '''),
-             Child('TrailingComma', kind='CommaToken', is_optional=True),
+         ]),
+
+    # SWIFT_ENABLE_TENSORFLOW
+    # The argument of '@differentiating(...)'.
+    # differentiating-attr-arguments ->
+    #     differentiable-attr-func-specifier # original
```

Comment:
This comment is incorrect:
```suggestion
    #     func-decl-name
```
I'll update it in my next PR.

Reply:
Fixed in https://github.com/apple/swift/pull/23350/files.